packages/react-core/src/components/Alert/Alert.tsx

```diff
@@ -46,6 +47,8 @@ export interface AlertProps extends Omit<React.HTMLProps<HTMLDivElement>, 'actio
   truncateTitle?: number;
   /** Position of the tooltip which is displayed if text is truncated */
   tooltipPosition?: 'auto' | 'top' | 'bottom' | 'left' | 'right';
+  /** Set a custom icon to the Alert. If not set the icon is set according to the variant */
+  customIcon?: React.ComponentClass<SVGIconProps>;
```

Comment:
To piggyback on @jschuler 's change of mind let's make this `React.ReactNode` to allow anything.

Reply:
That was my initial thought as well. That is what we do elsewhere.  
@boaz was there a reason for the `React.ComponentClass<SVGIconProps>` type here?

Reply:
The `createIcon` method returns `React.ComponentClass<SVGIconProps>` and I assumed we want the user to pass only icons.
If this is not an issue then I am OK with going with the `React.ReacNode` type.

Reply:
It's not an issue. We want to offer the user the most flexibility.